### DISCUSSION TITLE
Upgradeable contracts refresher missing

### DISCUSSION THREAD

**Discussion by AslanBlack** [Cyfrin/security-and-auditing-full-course-s23#135] [General]
Hi there! :wave: 

First of all, let me say I enjoy the courses a lot. All that great learning material for free, it's incredible. Thank you to the Cyfrin team for that. But I have a little problem with one of the lessons. 

In the video for Lesson 12 of the first section - [Upgradable Contrat ](https://updraft.cyfrin.io/courses/security/review/upgradeable-contracts?lesson_format=video)(refresher) - @PatrickAlphaC says that we're gonna watch some clips from the Foundry full course about proxies and delegate calls, then the video ends. However, the next lesson/video is about Self Destruct. 

Also, the written version of lesson 12 seems to imply that this video actually covers the subject :point_down: 

![Screenshot from 2024-02-27 11-19-29](https://github.com/Cyfrin/security-and-auditing-full-course-s23/assets/10978096/1769afec-c027-4c4f-9a59-721e0d237704)

I'm I missing something here? 

Cheers, 

Kev

**Comment by EngrPips** [comment]
Hey @AslanBlack  Yeah you are correct that in the video Patrick said `we are gonna watch some video about proxies and code small examples on how to do it`  but that was not actually done. Some stuffs are excerpt from previously created course so that might have not been properly fitted In. But there is a detailed explanation  in the foundry-full-course that covers how proxies and delegate call works.

**Reply by AslanBlack** [reply]
Thanks for your answer here and on Discord.

**Comment by allwin199** [comment]
@AslanBlack check [Proxies](https://github.com/Cyfrin/foundry-full-course-f23?tab=readme-ov-file#lesson-13-foundry-upgrades)

**Reply by AslanBlack** [reply]
Yep. That's what I ended up doing. Thanks a lot. 👍 
 I just thought I would mention it anyway in case it could be useful.
